### PR TITLE
Messaging Interface Fix: Handle old & new signatures

### PR DESCRIPTION
**Pull request by huzaif-plivo**

### CODE REVIEW

File: src/Plivo/Resource/Message/MessageInterface.cs [hunk 1]
```diff
@@ -176,6 +176,163 @@ public MessageInterface(HttpClient client) : base(client)
...
+        /// <param name="trackable">trackable.</param>
+        /// <param name="powerpackUUID">powerpackUUID</param>
+        ///<param name="media_urls">media_urls</param>
+        ///<param name="media_ids">media_ids</param>
+        public MessageCreateResponse Create(
+            string dst, string text = null, string src = null, string type = null,
+            string url = null, string method = null, bool? log = null, bool? trackable = null, string powerpack_uuid = null, string[] media_urls = null, string[] media_ids = null)
+        {
+
+            string _dst = string.Join("<", dst);
```
Comment (narayana-plivo) [inline src/Plivo/Resource/Message/MessageInterface.cs:203]: why do we need join here?

**Review by narayana-plivo — COMMENTED** [review]
I cant see respective unit test case change
please change specific field, dont change whole function when your fix the bug

**Comment by huzaif-plivo** [comment]
> I cant see respective unit test case change
> please change specific field, dont change whole function when your fix the bug

Requested changes have been made.

File: tests_netcore/Plivo.NetCore.Test/Resources/TestMessage.cs [hunk 2]
```diff
@@ -39,7 +39,25 @@ public void TestMessageCreate()
...
+            Assert.Empty(
+               ComparisonUtilities.Compare(
+                   response,
+                   Api.Message.Create(dst: "+919898989898<+919090909090",
+                      text: "textext",powerpack_uuid:"asdasd-asdasd-asdasd21-asd")));
+
+            Assert.Empty(
+               ComparisonUtilities.Compare(
+                   response,
+                   Api.Message.Create(dst: new List<string>() { "+919898989898", "+919090909090" },
```
Comment (narayana-plivo) [inline tests_netcore/Plivo.NetCore.Test/Resources/TestMessage.cs:59]: As per your change, dst is a string param, but  here your passing list<string> as dst param. is this working fine?

File: tests/Plivo.Test/Resources/TestMessage.cs [hunk 3]
```diff
@@ -29,21 +28,39 @@ public void TestMessageCreate()
...
+            Assert.Empty(
+               ComparisonUtilities.Compare(
+                   response,
+                   Api.Message.Create(dst: "+919898989898<+919090909090",
+                      text: "textext",powerpack_uuid:"asdasd-asdasd-asdasd21-asd")));
+
+            Assert.Empty(
+               ComparisonUtilities.Compare(
+                   response,
+                   Api.Message.Create(dst: new List<string>() { "+919898989898", "+919090909090" },
```
Comment (narayana-plivo) [inline tests/Plivo.Test/Resources/TestMessage.cs:59]: same here

File: tests_netcore/Plivo.NetCore.Test/Resources/TestMessage.cs (hunk 2 above)
Comment (huzaif-plivo) [reply tests_netcore/Plivo.NetCore.Test/Resources/TestMessage.cs:59]: Yes, that's the idea to support both String and List.

File: tests/Plivo.Test/Resources/TestMessage.cs (hunk 3 above)
Comment (huzaif-plivo) [reply tests/Plivo.Test/Resources/TestMessage.cs:59]: Yes, that's the idea to support both String and List.

File: src/Plivo/Resource/Message/MessageInterface.cs [hunk 4]
```diff
@@ -176,6 +176,161 @@ public MessageInterface(HttpClient client) : base(client)
...
+        /// <param name="text">Text.</param>
+        /// <param name="type">Type.</param>
+        /// <param name="url">URL.</param>
+        /// <param name="method">Method.</param>
+        /// <param name="log">Log.</param>
+        /// <param name="trackable">trackable.</param>
+        /// <param name="powerpackUUID">powerpackUUID</param>
+        ///<param name="media_urls">media_urls</param>
+        ///<param name="media_ids">media_ids</param>
+        public MessageCreateResponse Create(
```
Comment (narayana-plivo) [inline src/Plivo/Resource/Message/MessageInterface.cs:198]: can we handle both string or List in same function? Lot of repeated duplicate code here

File: tests/Plivo.Test/Resources/TestMessage.cs [hunk 5]
```diff
@@ -1,16 +1,15 @@
 using System.Collections.Generic;
-using NUnit.Framework;
+using Xunit;
```
Comment (narayana-plivo) [inline tests/Plivo.Test/Resources/TestMessage.cs:2]: if your using Xunit, please use it in all places. Not good to keep two framework

Comment (huzaif-plivo) [reply tests/Plivo.Test/Resources/TestMessage.cs]: Fixed

**Review by narayana-plivo — APPROVED** [review]

**Pull request merged**